liquidprompt

```diff
@@ -1397,9 +1424,34 @@ _lp_kubernetes_context() {
 _lp_kubernetes_context_color() {
     _lp_kubernetes_context || return "$?"
 
-    lp_kubernetes_context_color="[${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}]"
+    lp_kubernetes_context_color="${LP_MARK_KUBECONTEXT}${LP_COLOR_KUBECONTEXT}${lp_kubernetes_context}${lp_kubernetes_namespace+:}${lp_kubernetes_namespace-}${NO_COL}"
 }
 
+_lp_cmake() {
+    (( LP_ENABLE_CMAKE )) || return 2
+    [[ ! -f CMakeCache.txt ]] && return 1
+
+    local cmake_compiler
+    IFS="=" read -r _ cmake_compiler <<<"$(grep COMPILER:FILEPATH CMakeCache.txt)"
+    cmake_compiler="${cmake_compiler##*/}" # Only the part after the last slash.
+    _lp_hash_color "$cmake_compiler"
+    lp_cmake_compiler="$lp_hash_color"
+
+    local cmake_generator
+    IFS="=" read -r _ cmake_generator <<<"$(grep CMAKE_GENERATOR:INTERNAL CMakeCache.txt)"
+    # Shorten: Makefiles -> Make, and Visual Studio -> VS.
+    cmake_generator="${cmake_generator/%Makefiles/Make}"
+    cmake_generator="${cmake_generator/#Visual Studio/VS}"
+    _lp_hash_color "$cmake_generator"
+    lp_cmake_generator="$lp_hash_color"
+
+    local cmake_buildtype
+    IFS="=" read -r _ cmake_buildtype <<<"$(grep CMAKE_BUILD_TYPE:STRING CMakeCache.txt)"
```

Comment:
This code is nice and clean, but calling a subshell 3 times and opening the same file also 3 times isn't the fastest.

I already did some crazy `grep` replacement in the python env:
https://github.com/nojhan/liquidprompt/blob/42dbe9ce08058fbb60d9bfef91625661c36d62b2/liquidprompt#L1235-L1245
Not saying you must, but something to consider.